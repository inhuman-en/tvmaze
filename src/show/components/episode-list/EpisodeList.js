import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { format as formatDate } from 'date-fns';
import './EpisodeList.scss';

const renderDate = (date) => formatDate(date, 'LLL d, y');

// NOTE: this list can be virtualized if beeded
const renderEpisodeItem = (episode) => (
  <tr key={episode.id} className="episode-list__row">
    <td>
      <span className="episode-list__row__number">
        {episode.season}x{episode.number}:
      </span>
    </td>
    <td>
      <Link to={`/episode/${episode.id}`} className="episode-list__row__episode-name">
        {episode.name}
      </Link>
    </td>
    <td className="episode-list__row__episode-date">{renderDate(new Date(episode.airdate))}</td>
  </tr>
);

const EpisodeList = ({ data }) => (
  <table className="episode-list">
    <thead className="episode-list__header">
      <tr className="episode-list__row">
        <th colSpan="2">Episode Name</th>
        <th>Airdate</th>
      </tr>
    </thead>
    <tbody className="episode-list__body">{data.map(renderEpisodeItem)}</tbody>
  </table>
);

EpisodeList.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      season: PropTypes.number.isRequired,
      number: PropTypes.number.isRequired,
      id: PropTypes.number.isRequired,
      airdate: PropTypes.string.isRequired,
    }),
  ),
};

export default EpisodeList;
