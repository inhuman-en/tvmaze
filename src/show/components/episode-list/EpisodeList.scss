@import 'style/variables';
@import 'style/mixins';

$row__height: 40px;

.episode-list {
  width: 100%;

  &__row {
    height: $row__height;
    text-align: left;
    & td,
    & th {
      padding-left: 10px;
    }

    &__number {
      font-weight: $semibold;
    }

    &__episode-name {
      color: $color__brand;
      text-decoration: underline;
    }

    &__episode-date {
      font-weight: $bold;
      width: 130px;
    }
  }

  &__header > &__row {
    background-color: $color__bg__header;
  }

  tbody > &__row:nth-child(odd) {
    background-color: $color__bg;
  }
}
