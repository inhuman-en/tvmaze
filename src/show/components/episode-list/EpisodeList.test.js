import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import EpisodeList from './EpisodeList';

configure({ adapter: new Adapter() });

describe('EpisodeList component', () => {
  const mockProps = {
    data: [
      { id: 1, number: 2, season: 3, name: 'episode 3x2', airdate: '2016-04-04' },
      { id: 1, number: 3, season: 3, name: 'episode 3x3', airdate: '2016-05-04' },
    ],
  };
  it('should render', () => {
    const result = toJson(shallow(<EpisodeList {...mockProps} />));
    expect(result).toMatchSnapshot();
  });
});
