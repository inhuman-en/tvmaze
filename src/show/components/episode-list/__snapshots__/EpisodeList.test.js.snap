// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeList component should render 1`] = `
<table
  className="episode-list"
>
  <thead
    className="episode-list__header"
  >
    <tr
      className="episode-list__row"
    >
      <th>
        Episode Name
      </th>
      <th>
        Airdate
      </th>
    </tr>
  </thead>
  <tbody
    className="episode-list__body"
  >
    <tr
      className="episode-list__row"
      key="1"
    >
      <td>
        <span
          className="episode-list__row__number"
        >
          3
          x
          2
          :
        </span>
        <Link
          className="episode-list__row__episode-name"
          to="/episode/1"
        >
          episode 3x2
        </Link>
      </td>
      <td>
        2016-04-04T00:00:00.000Z
      </td>
    </tr>
    <tr
      className="episode-list__row"
      key="1"
    >
      <td>
        <span
          className="episode-list__row__number"
        >
          3
          x
          3
          :
        </span>
        <Link
          className="episode-list__row__episode-name"
          to="/episode/1"
        >
          episode 3x3
        </Link>
      </td>
      <td>
        2016-05-04T00:00:00.000Z
      </td>
    </tr>
  </tbody>
</table>
`;
